src/Xenial.Tasty.Cli/Commands/InteractiveCommand.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using StreamJsonRpc;
+using Xenial.Delicious.Protocols;
+using static SimpleExec.Command;
+using static Xenial.Delicious.Utils.PromiseHelper;
+
+namespace Xenial.Delicious.Cli.Commands
+{
+    public static class InteractiveCommand
+    {
+        public static async Task<int> Interactive(string project, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var path = Path.GetFullPath(project);
+                if (!string.IsNullOrEmpty(path) && Directory.Exists(path))
```

Comment:
Directory.Exists is safe and will return false on null or empty.

Reply:
This is true, but MS did not annotate the `Directory.Exists` with the `NotNull` annotation. So a usage of path fails later.